File: app/src/main/java/com/konektedi/vs/Home/Categories/CategoriesViewModel.java
package com.konektedi.vs.Home.Categories;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;


public class CategoriesViewModel extends AndroidViewModel {

    private LiveData<List<CategoriesModel>> categories;
    private CategoriesRepository categoriesRepository = new CategoriesRepository();


    public CategoriesViewModel(@NonNull Application application) {
        super(application);
    }

    public LiveData<List<CategoriesModel>> getAllCategories(String election_id) {
        categories = categoriesRepository.getCategories(election_id);
        return categories;
    }
}

File: app/src/main/java/com/konektedi/vs/categories/CategoriesViewModel.kt
package com.konektedi.vs.categories

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import com.konektedi.vs.utilities.models.Category
import com.konektedi.vs.utilities.models.Stat


class CategoriesViewModel : ViewModel() {
    private val repo = CategoriesRepository()

    val networkState = repo.mNetworkState

    fun getCategories(election_id: Int): LiveData<List<Category>> {
        return repo.getCategories(election_id)
    }

    fun getStats(election_id: Int): LiveData<List<Stat>> {
        return repo.getStats(election_id)
    }

}